src/websocket/index.ts

```diff
@@ -21,7 +21,22 @@ interface Request {
   params: Params
 }
 
+// Add connection counter
+let activeConnections = 0
+
 export const onConnection = async (socket: WebSocket.WebSocket): Promise<void> => {
+  // Check max connections limit
+  if (activeConnections >= CONFIG.websocket.maxConnections) {
+    socket.close(1003, 'Server busy. Please try again later.')
```

Comment:
will this properly clean up all the associated subscriptions?

Reply:
Here we are rejecting the connection before any subscriptions could be created, so yes that's handled :)